### ISSUE TITLE
Migrate to Apollo Client 3.0

### ISSUE THREAD

**Issue by ggwadera** [garageScript/c0d3-app#468]
All the Apollo Client packages have been merged into a single one for version 3. This means that all the packages we use from Apollo are now included within `@apollo/client`. Fortunately, they have published a [migration guide](https://www.apollographql.com/docs/react/migrating/apollo-client-3-migration/), where it says that there have been breaking changes from version 2. While there's a chance that nothing breaks on our end, I opted to leave this update out of the previous updates PR (#464) because it may involve a little more work.

On this same subject, I noticed we use [next-with-apollo](https://github.com/lfades/next-with-apollo) to provide a HOC for our components. While this is useful, it might also be not necessary, as I found this [example with Apollo from Vercel](https://github.com/vercel/next.js/tree/canary/examples/with-apollo) where Next is integrated with a GraphQL back-end using the Apollo client.

### To do

- [ ] Remove all current Apollo client packages
- [ ] Install only `@apollo/client`
- [ ] Update all the Apollo client imports
- [ ] Check if there's breaking changes (tests, type-check, check everything on localhost, etc...)
  - [ ] Tests
  - [ ] Type-check
  - [ ] Check app on localhost to see if it's working correctly
  - This includes checking if `next-with-apollo` is working correctly, as there is an [open issue](https://github.com/lfades/next-with-apollo/issues/147) about supporting Apollo Client v3.
- [ ] If something breaks, fix it

**Comment by Ulisseus** [comment]
I guess we're lucky since nothing broke on our end, I just passed over all files with the given transform. I believe only index.tsx needed one small fix. 
App is working locally, I can login as newbie/admin and view curriculum. Can submit challenges and review them. 
Some tests are broken though. Transform doesn't update mocks so you need to change `-jest.mock('@apollo/react-hooks')` to `+jest.mock('@apollo/client')`.
Everything under `__tests__/storyshots.test.js` (how can I update storybook snapshots?). And `__tests__/pages/[lesson].test.js` complains type error.

**Comment by ggwadera** [comment]
Good to know that nothing major breaks :)

The type errors may come from the `next-with-apollo` HOC, as it's mentioned in the linked issue. I believe there's nothing more we can do there besides ignoring them with `// @ts-ignore`.

The other choice is going without `next-with-apollo` as shown in the example from Vercel.

**Comment by Ulisseus** [comment]
Turns out these errors came from `ChallengeMaterial` component: 
```
 const challengesWithSubmissionData: ChallengeSubmissionData[] = challenges
    .sort((a, b) => a.order - b.order)
    .map((challenge: Challenge) => {
      const submission = userSubmissionsObject[challenge.id] || {}
      return {
        ...challenge,
        status: submission.status || 'unsubmitted',
        submission
      }
    })
   ```
 I am not sure why it didn't cause problems before but creating copy of challenges before sort-mapping prevented these errors. 
   `next-with-apollo` actually works fine despite not being updated for newer apollo client. Except for types of course but I was forced to use `@ts-ignore` only once so it should be fine. 
   Should I open a new PR?

**Comment by ggwadera** [comment]
> Should I open a new PR?

Sure, go ahead and open one! We can work out the issues then.
